src/qt/balancesdialog.cpp

```diff
@@ -150,18 +107,23 @@ BalancesDialog::BalancesDialog(QWidget *parent) :
     connect(balancesCopyAvailableAmountAction, SIGNAL(triggered()), this, SLOT(balancesCopyCol3()));
 }
 
+BalancesDialog::~BalancesDialog()
+{
+    delete ui;
+}
+
 void BalancesDialog::setClientModel(ClientModel *model)
 {
+    this->clientModel = model;
     if (model != NULL) {
-        this->clientModel = model;
         connect(model, SIGNAL(refreshOmniState()), this, SLOT(balancesUpdated()));
     }
 }
 
```

Comment:
I mirrored Bitcoin Core here, but I guess it comes down to be able to `delete` a model, in which case the `clientModel` is likely set to `NULL` before doing so.